coalib/settings/FunctionMetadata.py

```diff
@@ -160,3 +160,72 @@ def from_function(cls, func, omit=frozenset()):
                    non_optional_params=non_optional_params,
                    optional_params=optional_params,
                    omit=omit)
+
+    @classmethod
+    def merge(cls, *metadatas):
+        """
+        Merges signatures of ``FunctionMetadata`` objects.
+
+        Parameter (either optional or non-optional) and non-parameter
+        descriptions are merged from left to right, meaning the right hand
+        metadata overrides the left hand one.
+
+        >>> def a(x):
+        ...     '''
+        ...     desc of *a*
+        ...     :param x: x of a
```

Comment:
can you add another param here to show that it not only takes values from the `b` function?